pkg/siteconfig/clusterinstance_test.go

```diff
@@ -821,72 +821,14 @@ func TestClusterInstanceExists(t *testing.T) {
 	}
 }
 
-func TestClusterInstanceValidate(t *testing.T) {
-	testCases := []struct {
-		builderNil    bool
-		definitionNil bool
-		apiClientNil  bool
-		expectedError string
-	}{
-		{
-			builderNil:    true,
-			definitionNil: false,
-			apiClientNil:  false,
-			expectedError: "error: received nil ClusterInstance builder",
-		},
-		{
-			builderNil:    false,
-			definitionNil: true,
-			apiClientNil:  false,
-			expectedError: "can not redefine the undefined ClusterInstance",
-		},
-		{
-			builderNil:    false,
-			definitionNil: false,
-			apiClientNil:  true,
-			expectedError: "ClusterInstance builder cannot have nil apiClient",
-		},
-		{
-			builderNil:    false,
-			definitionNil: false,
-			apiClientNil:  false,
-			expectedError: "",
-		},
-	}
-
-	for _, testCase := range testCases {
-		testBuilder := generateClusterInstanceBuilderWithFakeObjects([]runtime.Object{})
-
-		if testCase.builderNil {
-			testBuilder = nil
-		}
-
-		if testCase.definitionNil {
-			testBuilder.Definition = nil
-		}
-
-		if testCase.apiClientNil {
-			testBuilder.apiClient = nil
-		}
+func generateClusterInstanceBuilderWithFakeObjects(objects []runtime.Object) *CIBuilder {
+	builder := NewCIBuilder(clients.GetTestClients(
```

Comment:
This is one limitation of this approach. Since we initialize the commonbuilder only during NewBuilder and Pull functions, we have to generate test builders either through the NewBuilder function or manually initialize builder.CommonBuilder

Reply:
For the manual step, could we create a `CreateTestCommbonBuilder` or something similar to the way we do `clients.GetTestClients`?